### ISSUE TITLE
learner.save_model gives KeyError while saving tokenizer/vocab file

### ISSUE THREAD

**Issue by mohammedayub44** [kaushaltrivedi/fast-bert#200]
I'm trying to run the multilabel classification model and while saving the model it give me an error on vocab file 
`learner.save_model()`
gives below error:
![image](https://user-images.githubusercontent.com/7122670/79359690-e5cf6500-7f10-11ea-937b-e26c1eb2967c.png)

Is this because I have not specified some path or because I'm not using a pretrained model path from local as in sample notebook.

My learner config is as below:
![image](https://user-images.githubusercontent.com/7122670/79359911-2cbd5a80-7f11-11ea-899f-ecd6155f94d5.png)

DataBunchConfig as below:
![image](https://user-images.githubusercontent.com/7122670/79359979-465ea200-7f11-11ea-9842-5bb2ee66af66.png)

Any help appreciated. Thanks!

**Comment by aaronbriel** [comment]
You need to pass save_model a Path to the directory you wish to save it to.

**Comment by mohammedayub44** [comment]
@aaronbriel  I did try passing path to the folder and I get the same error. 
Is it compulsory to pass the path ? If I don't pass any path to this function (as shown in sample notebook), I see that its picking up the "OUTPUTDIR" location (maybe from the learner object) and automatically creates "model_output" folder..it writes the following 4 files -
`pytorch_model.bin`
`special_tokens_map.json`
`config.json`
`tokenizer_config.json`

But fails to create `vocab.txt` as mentioned in documentation. Do you see the same issue on your end ?

Thanks !

**Comment by aaronbriel** [comment]
I am not seeing this issue. What is the actual error?

**Comment by mohammedayub44** [comment]
It just shows `KeyError` , which is strange. 
I verified my tokenizer name , model name etc. everything works fine except saving the model.

**Comment by mohammedayub44** [comment]
@aaronbriel  Here is the [Dropbox link](https://www.dropbox.com/sh/qs82eilcxskk0hv/AAAW44zuTCovhIXC-Lc40RqMa?dl=0) which contains the Python Notebook file along with data and labels folder. (training for only one epoch)
If you could let me if this runs and saves model successfully on your machine. (or if any errors) 

Appreciate the help!

**Comment by vivekam101** [comment]
I also faced same issue trying out example @
https://github.com/kaushaltrivedi/fast-bert/blob/master/test/multi_class.ipynb.

Please let us know.. Thanks a ton..

**Comment by yyn19951228** [comment]
same to me, I also have this issue.

**Comment by aaronbriel** [comment]
I meant to show a more complete stack trace as opposed to only KeyError or TypeError. That would be more helpful.

**Comment by vivekam101** [comment]
![image](https://user-images.githubusercontent.com/5806230/79711416-61505e00-82e5-11ea-8591-4c1c20d925d2.png)

**Comment by aaronbriel** [comment]
Sorry about the delay. I was indeed able to replicate the error with my current implementation using the latest fast-bert:
` File "/home/bert/.venv/lib/python3.6/site-packages/fast_bert/learner_util.py", line 128, in save_model
    self.data.tokenizer.save_pretrained(path)
  File "/home/bert/.venv/lib/python3.6/site-packages/transformers/tokenization_utils.py", line 605, in save_pretrained
    vocab_files = self.save_vocabulary(save_directory)
  File "/home/bert/.venv/lib/python3.6/site-packages/transformers/tokenization_utils.py", line 1994, in save_vocabulary
    files = self._tokenizer.save(save_directory)
  File "/home/bert/.venv/lib/python3.6/site-packages/tokenizers/implementations/base_tokenizer.py", line 222, in
 save
    return self._tokenizer.model.save(directory, name=name)
TypeError`

**Comment by yyn19951228** [comment]
So can we have any way to work around of this issue? Or We could use an old version?

**Comment by aaronbriel** [comment]
I wasn't seeing this before the last build, so you could try with that. I was going to look into it more tomorrow.

**Comment by yyn19951228** [comment]
Cool, thank you very much!

**Comment by mohammedayub44** [comment]
@aaronbriel  sorry for late reply. Thanks for taking a look into it. 
Looking forward to your solution.

**Comment by aaronbriel** [comment]
FYI, fixed in https://github.com/kaushaltrivedi/fast-bert/pull/205

**Comment by mohammedayub44** [comment]
Thanks @aaronbriel  Will check it out. 
Closing this for now.

**Comment by sidPN** [comment]
I'm still facing this same issue. Even with the latest fix.

**Comment by aaronbriel** [comment]
@kaushaltrivedi did you publish? The code is in place but fast-bert has not yet updated in pypi.

@sidPN you can use my fork until the fast-bert library is updated (git://github.com/aaronbriel/fast-bert@master#egg=fast_bert) or just download it locally.
